MINIFICPP-2746 Allow python processors to run in parallel (#2135)
apache/nifi-minifi-cpp: extensions/python/PYTHON.md

    return len(self.content)
```

> By default the MiNiFi C++ style native python processors are executed with multiple threads in parallel, but it is possible to set the processor to be single threaded by calling the `setSingleThreaded()` method in the processor while initializing. When a processor is set to be single threaded, only one thread will execute the processor, and setting the max concurrent tasks to more than 1 in the flow configuration will not have any effect.

Reviewer: more nitpicking:
```suggestion
By default, the MiNiFi C++ style native python processors are executed on multiple threads in parallel if 'max concurrent tasks' is set to more than 1 in the flow configuration, but it is possible to set the processor to be single threaded by calling the `setSingleThreaded()` method in the processor while initializing. When a processor is set to be single threaded, only one thread will execute the processor, and setting 'max concurrent tasks' to more than 1 will not have any effect.
```

Author: Updated in https://github.com/apache/nifi-minifi-cpp/pull/2135/commits/6b11275afa0753c91c841ae4310a48b882a48565
